common/scala/src/main/scala/whisk/connector/kafka/KafkaConsumerConnector.scala

```diff
@@ -182,10 +185,11 @@ class KafkaConsumerConnector(
       blocking {
         if (offset > 0) {
           val topicAndPartition = new TopicPartition(topic, 0)
-          consumer.endOffsets(Set(topicAndPartition).asJava).asScala.get(topicAndPartition).foreach { endOffset =>
-            // endOffset could lag behind the offset reported by the consumer internally resulting in negative numbers
-            val queueSize = (endOffset - offset).max(0)
-            MetricEmitter.emitHistogramMetric(queueMetric, queueSize)
+          synchronized(consumer.endOffsets(Set(topicAndPartition).asJava)).asScala.get(topicAndPartition).foreach {
```

Comment:
Here you’re synchronizing again given the outer sync. Is this because of the blocking block?

Reply:
I don't see what you mean by "synchronizing again given the outer sync"? There is no outer sync for this scheduled task. The `blocking` block is just a hint to the underlying thread pool that this Future may block such that the creation of a new thread is desirable to not block too many threads.

Reply:
Referring to Line 159.

Reply:
This line also puzzled me at the first sight. If you expand the section of compared lines in this diff view, you will see that the `synchronized` in line 159 belongs to method `recreateConsumer()` while the `synchronized` you commented on belongs to a totally different scope in the scheduled task.